### ISSUE TITLE
rev -> revhead conversion can happen while restrained, in crit

### ISSUE THREAD

**Issue by PKPenguin321** [tgstation/tgstation#21298]
Had this happen to a player the other day where they were in handcuffs and while a loyalty implant was in the process of being injected into their body another station head signed up, creating a new revhead out of a rev which happened to be that player. Implant was rejected and they died because of it, cucking both that player out of their round and the rest of the revolution out of a revhead.

Tl;dr: Midround rev to revhead conversion needs to respect restrained/stunned/in crit statuses

**Comment by KingPhilipIII** [comment]
I've had this happen before. Arrested for an unrelated charge. Didn't know it was revs. I was stripped and searched. I got promoted while in cuffs being held. An engineering bag materialized with the rev stuff. 

Sec didn't notice and I got released shortly later.
